esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -161,14 +258,29 @@ void WUDetailsVisitor::noteHint(const char * kind, const char * value)
     EspWUResponseProperties.append(*EspWUResponseProperty.getClear());
 }
 
+// Get StatisticKind or WuAttr from property name and return true
+// When neither possible, return false.
+static bool getPropertyIdFromName(const char *propName, StatisticKind & sk, WuAttr & wa )
+{
+    sk = queryStatisticKind(propName, StKindNone);
```

Comment:
The original code is actually correct: it only attempts to initialize wa if it can't find a valid name as StatisticKind.

However, I have refactored to make the code clearer.

Reply:
I agree the code is correct for the current usage, but if you happened to call the same function in a different content you could get confusing results.

Reply:
You should either cleanly initialise wa (preferred) or document in the function description that you need to check sk before wa.